active_learning/active_learning.py

```diff
@@ -1,21 +1,29 @@
-from util.datastructures import MetaPath
+from util.datastructures import MetaPath, MetaPathRatingGraph
 
 # mock data
 mock_paths = [MetaPath([1, 2], [3]), MetaPath([1, 2, 1], [3, 4]), MetaPath([1, 3, 2], [1, 1]),
               MetaPath([1, 3, 2], [1, 1]), MetaPath([1, 3, 2], [1, 1])]
 
 # algorithm types
-NO_RATING = 'no_rating'
+NO_USER_FEEDBACK = 'no_user_feedback'
 BASELINE = 'baseline'
 
 
 class Oracle:
     """
     Abstract class for the Oracle the Active Learner interacts with. 
     """
+    number_of_calls = None
+
+    def __init__(self):
+        self.number_of_calls = 0
 
     # TODO Think about whether exclusion is own class or 0
     def rate_meta_path(self, meta_path: MetaPath) -> float:
+        self.number_of_calls += 1
+        return self._rate_meta_path(meta_path)
+
+    def _rate_meta_path(self, meta_path: MetaPath) -> float:
         raise NotImplementedError("Should have implemented this.")
```

Comment:
Maybe put "Subclass should have implemented this"? Otherwise I might think this class is in progress.

Reply:
There is not such error. If its important to you, we could define one in util, but what I did seems to be the standard method